PR - Currency crosses - Issue 29

davetromp opened the pull request:
Implementation of currencies retrieval for all currencies listed at https://www.investing.com/currencies/streaming-forex-rates-majors

alvarob96 commented on tests/test_investpy.py:
```diff
@@ -445,9 +445,27 @@ def test_investpy_indices():
     retrieve_index_countries(test_mode=True)
 
 
+def test_investpy_currencies():
```
Both `tests/` for errors and successes should be implemented in order to keep codecov percentage over 90%. So on this test file should be improved or kept simple, but notifying the owner that before releasing new version the tests need to be updated.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -288,9 +301,11 @@ def get_recent_data(equity, country, as_json=False, order='ascending', debug=Fal
     req = requests.post(url, headers=head, data=params)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -305,9 +320,11 @@ def get_recent_data(equity, country, as_json=False, order='ascending', debug=Fal
                 info.append(nested_.text_content())
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -316,11 +333,14 @@ def get_recent_data(equity, country, as_json=False, order='ascending', debug=Fal
             stock_volume = 0
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -341,7 +361,8 @@ def get_recent_data(equity, country, as_json=False, order='ascending', debug=Fal
 
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -416,41 +437,49 @@ def get_historical_data(equity, country, from_date, to_date, as_json=False, orde
     """
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -202,46 +202,56 @@ def get_recent_data(equity, country, as_json=False, order='ascending', debug=Fal
     """
 
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -253,11 +263,14 @@ def get_recent_data(equity, country, as_json=False, order='ascending', debug=Fal
 
     logger.info('Searching introduced equity on Investing.com')
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -488,23 +517,28 @@ def get_historical_data(equity, country, from_date, to_date, as_json=False, orde
     resource_package = __name__
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -558,7 +595,8 @@ def get_historical_data(equity, country, from_date, to_date, as_json=False, orde
         req = requests.post(url, headers=head, data=params)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -578,25 +616,34 @@ def get_historical_data(equity, country, from_date, to_date, as_json=False, orde
                     if interval_counter < interval_limit:
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -675,49 +724,58 @@ def get_equity_company_profile(equity, country='spain', language='english'):
     }
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -734,7 +792,8 @@ def get_equity_company_profile(equity, country='spain', language='english'):
         req = requests.get(url, headers=head)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -3161,9 +3370,588 @@ def search_indices(by, value):
     search_result = indices.loc[indices['matches'] == True].copy()
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -3131,25 +3335,30 @@ def search_indices(by, value):
     available_search_fields = ['name', 'full_name']
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2480,9 +2682,11 @@ def search_etfs(by, value):
     search_result = etfs.loc[etfs['matches'] == True].copy()
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2450,25 +2647,30 @@ def search_etfs(by, value):
     available_search_fields = ['name', 'symbol']
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2387,25 +2578,31 @@ def get_etfs_overview(country, as_json=False):
     if path_:
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2370,12 +2559,14 @@ def get_etfs_overview(country, as_json=False):
     if unidecode.unidecode(country.lower()) not in get_etf_countries():
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2351,13 +2538,15 @@ def get_etfs_overview(country, as_json=False):
     """
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2306,13 +2491,15 @@ def get_etf_historical_data(etf, country, from_date, to_date, as_json=False, ord
 
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2278,15 +2458,20 @@ def get_etf_historical_data(etf, country, from_date, to_date, as_json=False, ord
                     if interval_counter < interval_limit:
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2254,9 +2432,11 @@ def get_etf_historical_data(etf, country, from_date, to_date, as_json=False, ord
         req = requests.post(url, headers=head, data=params)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2216,7 +2393,8 @@ def get_etf_historical_data(etf, country, from_date, to_date, as_json=False, ord
     id_ = etfs.loc[(etfs['name'].str.lower() == etf).idxmax(), 'id']
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2184,23 +2358,26 @@ def get_etf_historical_data(etf, country, from_date, to_date, as_json=False, ord
     resource_package = __name__
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2112,41 +2278,49 @@ def get_etf_historical_data(etf, country, from_date, to_date, as_json=False, ord
     """
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2036,7 +2201,8 @@ def get_etf_recent_data(etf, country, as_json=False, order='ascending', debug=Fa
 
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -2011,15 +2173,18 @@ def get_etf_recent_data(etf, country, as_json=False, order='ascending', debug=Fa
                 info.append(nested_.text_content())
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1994,9 +2154,11 @@ def get_etf_recent_data(etf, country, as_json=False, order='ascending', debug=Fa
     req = requests.post(url, headers=head, data=params)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1963,7 +2122,8 @@ def get_etf_recent_data(etf, country, as_json=False, order='ascending', debug=Fa
     id_ = etfs.loc[(etfs['name'].str.lower() == etf).idxmax(), 'id']
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1908,46 +2059,54 @@ def get_etf_recent_data(etf, country, as_json=False, order='ascending', debug=Fa
     """
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1676,37 +1820,44 @@ def search_funds(by, value):
     available_search_fields = ['name', 'symbol', 'issuer', 'isin']
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1547,93 +1662,122 @@ def get_fund_information(fund, country, as_json=False):
 
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1535,7 +1649,8 @@ def get_fund_information(fund, country, as_json=False):
     req = requests.get(url, headers=head)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1485,40 +1592,47 @@ def get_fund_information(fund, country, as_json=False):
     """
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1425,13 +1530,15 @@ def get_fund_historical_data(fund, country, from_date, to_date, as_json=False, o
 
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1396,17 +1495,23 @@ def get_fund_historical_data(fund, country, from_date, to_date, as_json=False, o
                     if interval_counter < interval_limit:
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1375,9 +1472,11 @@ def get_fund_historical_data(fund, country, from_date, to_date, as_json=False, o
         req = requests.post(url, headers=head, data=params)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1337,7 +1433,8 @@ def get_fund_historical_data(fund, country, from_date, to_date, as_json=False, o
     id_ = funds.loc[(funds['name'].str.lower() == fund).idxmax(), 'id']
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1305,23 +1397,27 @@ def get_fund_historical_data(fund, country, from_date, to_date, as_json=False, o
     resource_package = __name__
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1233,41 +1317,49 @@ def get_fund_historical_data(fund, country, from_date, to_date, as_json=False, o
     """
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1159,7 +1242,8 @@ def get_fund_recent_data(fund, country, as_json=False, order='ascending', debug=
 
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1133,9 +1214,11 @@ def get_fund_recent_data(fund, country, as_json=False, order='ascending', debug=
                 info.append(nested_.text_content())
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1116,9 +1195,11 @@ def get_fund_recent_data(fund, country, as_json=False, order='ascending', debug=
     req = requests.post(url, headers=head, data=params)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1085,7 +1163,8 @@ def get_fund_recent_data(fund, country, as_json=False, order='ascending', debug=
     id_ = funds.loc[(funds['name'].str.lower() == fund).idxmax(), 'id']
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -615,13 +662,15 @@ def get_historical_data(equity, country, from_date, to_date, as_json=False, orde
                              }
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -748,13 +807,15 @@ def get_equity_company_profile(equity, country='spain', language='english'):
 
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -771,7 +832,8 @@ def get_equity_company_profile(equity, country='spain', language='english'):
         req = requests.get(url, headers=head)
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -810,37 +872,44 @@ def search_equities(by, value):
     available_search_fields = ['name', 'full_name', 'isin']
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/__init__.py:
```diff
@@ -1030,46 +1099,55 @@ def get_fund_recent_data(fund, country, as_json=False, order='ascending', debug=
     """
```
All the changes done not related with `cross_currencies` should not be included since the previous code was properly formatted and the published changes are just modifications on how the data is tabulated or presented to the programmer. So **just the additions related to cross currencies data retrieval will be taken into consideration**, unless changes in any other function improves it.

alvarob96 commented on investpy/currency_crosses.py:
```diff
@@ -0,0 +1,248 @@
+        currencies = retrieve_currency_crosses()
+
+    if currencies is None:
+        raise IOError("ERR#0037: currencies not found or unable to retrieve.")
```
Error codes cannot be duplicated, so the error code should be formatted as `ERR#xxxx` and the code should be a different one for each kind of error.

alvarob96 commented on investpy/currency_crosses.py:
```diff
@@ -0,0 +1,248 @@
+        columns = currencies.columns.tolist()
+    else:
+        if not isinstance(columns, list):
+            raise ValueError("ERR#0020: specified columns argument is not a list, it can just be list type.")
```
Error codes cannot be duplicated, so the error code should be formatted as `ERR#xxxx` and the code should be a different one for each kind of error.

alvarob96 commented on investpy/currency_crosses.py:
```diff
@@ -0,0 +1,248 @@
+            raise ValueError("ERR#0020: specified columns argument is not a list, it can just be list type.")
+
+    if not all(column in currencies.columns.tolist() for column in columns):
+        raise ValueError("ERR#0023: specified columns does not exist, available columns are "
```
Error codes cannot be duplicated, so the error code should be formatted as `ERR#xxxx` and the code should be a different one for each kind of error.

alvarob96 commented on investpy/currency_crosses.py:
```diff
@@ -0,0 +1,248 @@
+
+
+def retrieve_currency_crosses(test_mode=False):
+    """
```
docstrings should be properly explained and checked before pulling the request.

alvarob96 commented on investpy/currency_crosses.py:
```diff
@@ -0,0 +1,248 @@
+
+
+def currency_crosses_as_df():
+    """
```
docstrings should be properly explained and checked before pulling the request.

alvarob96 commented on investpy/currency_crosses.py:
```diff
@@ -0,0 +1,248 @@
+
+
+def currency_crosses_as_list():
+    """
```
docstrings should be properly explained and checked before pulling the request.

alvarob96 commented on investpy/currency_crosses.py:
```diff
@@ -0,0 +1,248 @@
+
+
+def currency_crosses_as_dict(columns=None, as_json=False):
+    """
```
docstrings should be properly explained and checked before pulling the request.

alvarob96 commented on investpy/currency_crosses.py:
```diff
@@ -0,0 +1,248 @@
+    }
+
+    # url = "https://www.investing.com/currencies/single-currency-crosses"
+    url = "https://www.investing.com/currencies/streaming-forex-rates-majors"
```
This should be properly explained either on the code via comments or opening an issue related to it whenever the pull request is done.

alvarob96 commented:
They look like a lot of changes to make, but the main ones are just things to revert in [\_\_init\_\_.py](https://github.com/alvarob96/investpy/blob/master/investpy/__init__.py) file since they are not needed.

davetromp commented:
I have updated the docstrings, error codes and tests. I also removed the reformatting. Sorry about that. Also, the currency is no longer set to None, but to the destination currency of the currency cross. Let me know if there are any more changes needed. Also, please note that only the major currency crosses have been implemented with this PR.

davetromp commented:
Is there an overview of the error codes somewhere? I will be happy to fix them.
